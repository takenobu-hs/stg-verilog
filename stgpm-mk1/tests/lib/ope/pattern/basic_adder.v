

`define SP1_ADDER

  parameter DW = 32;


  /* test pattern */

  initial begin
    @(negedge rst);
    repeat(2) @(posedge clk);

    @(posedge clk);
    a0 = 'd1;
    a1 = 'd2;

    @(posedge clk);
    a0 = 32'h1234_5678;
    a1 = 32'h1111_1111;

    @(posedge clk);
    a0 = 32'hffff_fffe;
    a1 = 32'h0000_0001;

    /* carry */
    @(posedge clk);
    a0 = 32'hffff_ffff;
    a1 = 32'h0000_0001;

    repeat(5) @(posedge clk);
    halt();
  end


