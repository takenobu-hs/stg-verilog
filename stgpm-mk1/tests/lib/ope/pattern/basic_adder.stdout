monitor   :       0: xxxxxxxx xxxxxxxx  x xxxxxxxx
monitor   :      80: 00000001 00000002  0 00000003
monitor   :      90: 12345678 11111111  0 23456789
monitor   :     100: fffffffe 00000001  0 ffffffff
monitor   :     110: ffffffff 00000001  1 00000000
halt      : halt
